GRDB/Record/PersistableRecord+Encodable.swift

```diff
@@ -194,6 +214,92 @@ private struct PersistableRecordEncoder : Encoder {
     }
 }
 
+class ThrowingKeyedContainer<KeyType: CodingKey>: KeyedEncodingContainerProtocol {
+    let errorMessage: Error
+    var codingPath: [CodingKey] = []
+    
+    init(error: Error) {
+        errorMessage = error
+    }
+    
+    func encodeNil(forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: Bool, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: Int, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: Int8, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: Int16, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: Int32, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: Int64, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: UInt, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: UInt8, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: UInt16, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: UInt32, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: UInt64, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: Float, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: Double, forKey key: KeyType) throws { throw errorMessage }
+    func encode(_ value: String, forKey key: KeyType) throws { throw errorMessage }
+    func encode<T>(_ value: T, forKey key: KeyType) throws where T : Encodable { throw errorMessage }
+
+    func nestedContainer<NestedKey>(keyedBy keyType: NestedKey.Type, forKey key: KeyType) -> KeyedEncodingContainer<NestedKey> where NestedKey : CodingKey {
+        fatalError("Not implemented")
```

Comment:
I think we should return another throwing container here. Imagine a codable type that tries to access a nested container *before* it stores its properties and triggers one of our throwing `encode` methods above. Program will crash instead of switching to JSON encoding. This type should **never** fatalError, so that we can **always** switch to JSON. That's the **whole point**.

Reply:
Thanks @groue good point :-), commit c52b59f1 now returns another throwing container.